Linkage issue with the latest commit in main

Hey guys, 

I cannot compile with the option 2 - https://github.com/wsmoses/Polygeist#option-2-using-unified-llvm-mlir-clang-and-polygeist-build

The issue is in linking - 

```
[4352/4352] Linking CXX executable bin/mlir-clang
FAILED: bin/mlir-clang
: && /mnt/ccnas2/bdp/rz3515/cccad/opt/bin/g++ -fPIC -fno-semantic-interposition -fvisibility-inlines-hidden -Werror=date-time -Wall -Wextra -Wno-unused-parameter -Wwrite-strings -Wcast-qual -Wno-missing-field-initializers -pedantic -Wno-long-long -Wimplicit-fallthrough -Wno-maybe-uninitialized -Wno-class-memaccess -Wno-redundant-move -Wno-pessimizing-move -Wno-noexcept-type -Wnon-virtual-dtor -Wdelete-non-virtual-dtor -Wsuggest-override -Wno-comment -Wmisleading-indentation -fdiagnostics-color -fPIC -fno-semantic-interposition -fvisibility-inlines-hidden -Werror=date-time -Wall -Wextra -Wno-unused-parameter -Wwrite-strings -Wcast-qual -Wno-missing-field-initializers -pedantic -Wno-long-long -Wimplicit-fallthrough -Wno-maybe-uninitialized -Wno-class-memaccess -Wno-redundant-move -Wno-pessimizing-move -Wno-noexcept-type -Wnon-virtual-dtor -Wdelete-non-virtual-dtor -Wsuggest-override -Wno-comment -Wmisleading-indentation -fdiagnostics-color -g -Wl,-rpath-link,/mnt/ccnas2/bdp/rz3515/projects/Polygeist/build/./lib tools/polygeist/mlir-clang/CMakeFiles/mlir-clang.dir/__/llvm-project/clang/tools/driver/cc1_main.cpp.o tools/polygeist/mlir-clang/CMakeFiles/mlir-clang.dir/__/llvm-project/clang/tools/driver/cc1as_main.cpp.o tools/polygeist/mlir-clang/CMakeFiles/mlir-clang.dir/__/llvm-project/clang/tools/driver/cc1gen_reproducer_main.cpp.o tools/polygeist/mlir-clang/CMakeFiles/mlir-clang.dir/mlir-clang.cc.o tools/polygeist/mlir-clang/CMakeFiles/mlir-clang.dir/Lib/utils.cc.o tools/polygeist/mlir-clang/CMakeFiles/mlir-clang.dir/Lib/pragmaHandler.cc.o -o bin/mlir-clang  -Wl,-rpath,"\$ORIGIN/../lib"  -lpthread  lib/libMLIRSCFTransforms.a  lib/libMLIRPolygeist.a  lib/libMLIRSupport.a  lib/libMLIRIR.a  lib/libMLIRAnalysis.a  lib/libMLIRLLVMIR.a  lib/libMLIRNVVMIR.a  lib/libMLIROpenMP.a  lib/libMLIRGPUOps.a  lib/libMLIRTransforms.a  lib/libMLIRSCFToStandard.a  lib/libMLIRStandardToLLVM.a  lib/libMLIRAffineTransforms.a  lib/libMLIRAffineToStandard.a  lib/libMLIRMathToLLVM.a  lib/libMLIRTargetLLVMIRImport.a  lib/libMLIRPolygeistTransforms.a  lib/libMLIRLLVMToLLVMIRTranslation.a  lib/libMLIRSCFToOpenMP.a  lib/libMLIROpenMPToLLVM.a  lib/libMLIROpenMPToLLVMIRTranslation.a  lib/libclangAST.a  lib/libclangBasic.a  lib/libclangCodeGen.a  lib/libclangDriver.a  lib/libclangFrontend.a  lib/libclangFrontendTool.a  lib/libclangLex.a  lib/libclangSerialization.a  lib/libMLIRVectorToLLVM.a  lib/libMLIRArmNeon.a  lib/libMLIRArmSVETransforms.a  lib/libMLIRArmSVE.a  lib/libMLIRAMXTransforms.a  lib/libMLIRAMX.a  lib/libMLIRX86VectorTransforms.a  lib/libMLIRX86Vector.a  lib/libMLIRPolygeist.a  lib/libMLIRNVVMIR.a  lib/libMLIRGPUOps.a  lib/libMLIRDLTI.a  lib/libMLIRSCFToStandard.a  lib/libMLIRStandardOpsTransforms.a  lib/libMLIRStandardToLLVM.a  lib/libMLIRMemRefToLLVM.a  lib/libMLIRLLVMCommonConversion.a  lib/libMLIRTransforms.a  lib/libMLIRCopyOpInterface.a  lib/libMLIRVector.a  lib/libMLIRAffineUtils.a  lib/libMLIRTransformUtils.a  lib/libMLIRRewrite.a  lib/libMLIRPDLToPDLInterp.a  lib/libMLIRPDLInterp.a  lib/libMLIRPDL.a  lib/libMLIRLoopAnalysis.a  lib/libMLIRPresburger.a  lib/libMLIROpenMP.a  lib/libMLIRTargetLLVMIRExport.a  lib/libMLIRTranslation.a  lib/libMLIRLLVMIRTransforms.a  lib/libMLIRLLVMIR.a  lib/libMLIRPass.a  lib/libMLIRAnalysis.a  lib/libMLIRDataLayoutInterfaces.a  lib/libMLIRLinalg.a  lib/libMLIRAffine.a  lib/libMLIRSCF.a  lib/libMLIRMemRef.a  lib/libMLIRMemRefUtils.a  lib/libMLIRLoopLikeInterface.a  lib/libMLIRTensor.a  lib/libMLIRStandard.a  lib/libMLIRCallInterfaces.a  lib/libMLIRControlFlowInterfaces.a  lib/libMLIRVectorInterfaces.a  lib/libMLIRCastInterfaces.a  lib/libMLIRInferTypeOpInterface.a  lib/libMLIRSideEffectInterfaces.a  lib/libMLIRMath.a  lib/libMLIRDialect.a  lib/libMLIRDialectUtils.a  lib/libMLIRViewLikeInterface.a  lib/libMLIRParser.a  lib/libMLIRTilingInterface.a  lib/libMLIRIR.a  lib/libMLIRSupport.a  lib/libclangCodeGen.a  lib/libLLVMCoverage.a  lib/libLLVMLTO.a  lib/libLLVMExtensions.a  lib/libLLVMCodeGen.a  lib/libLLVMPasses.a  lib/libLLVMCoroutines.a  lib/libLLVMipo.a  lib/libLLVMBitWriter.a  lib/libLLVMIRReader.a  lib/libLLVMAsmParser.a  lib/libLLVMLinker.a  lib/libLLVMInstrumentation.a  lib/libLLVMObjCARCOpts.a  lib/libLLVMVectorize.a  lib/libLLVMScalarOpts.a  lib/libLLVMAggressiveInstCombine.a  lib/libLLVMInstCombine.a  lib/libLLVMTarget.a  lib/libclangRewriteFrontend.a  lib/libclangARCMigrate.a  lib/libclangStaticAnalyzerFrontend.a  lib/libclangStaticAnalyzerCheckers.a  lib/libclangStaticAnalyzerCore.a  lib/libclangCrossTU.a  lib/libclangIndex.a  lib/libclangFrontend.a  lib/libclangDriver.a  lib/libLLVMOption.a  lib/libclangParse.a  lib/libclangSerialization.a  lib/libclangSema.a  lib/libclangAnalysis.a  lib/libclangASTMatchers.a  lib/libclangEdit.a  lib/libclangAST.a  lib/libLLVMFrontendOpenMP.a  lib/libLLVMTransformUtils.a  lib/libLLVMAnalysis.a  lib/libLLVMObject.a  lib/libLLVMBitReader.a  lib/libLLVMMCParser.a  lib/libLLVMMC.a  lib/libLLVMDebugInfoCodeView.a  lib/libLLVMTextAPI.a  lib/libLLVMProfileData.a  lib/libLLVMCore.a  lib/libLLVMBinaryFormat.a  lib/libLLVMRemarks.a  lib/libLLVMBitstreamReader.a  lib/libclangFormat.a  lib/libclangToolingInclusions.a  lib/libclangToolingCore.a  lib/libclangRewrite.a  lib/libclangLex.a  lib/libclangBasic.a  lib/libLLVMSupport.a  -lrt  -ldl  -lpthread  -lm  /usr/lib64/libz.so  /usr/lib64/libtinfo.so  lib/libLLVMDemangle.a && :
tools/polygeist/mlir-clang/CMakeFiles/mlir-clang.dir/__/llvm-project/clang/tools/driver/cc1_main.cpp.o: In function `llvm::InitializeAllTargetInfos()':
/mnt/ccnas2/bdp/rz3515/projects/Polygeist/build/include/llvm/Config/Targets.def:26: undefined reference to `LLVMInitializeX86TargetInfo'
tools/polygeist/mlir-clang/CMakeFiles/mlir-clang.dir/__/llvm-project/clang/tools/driver/cc1_main.cpp.o: In function `llvm::InitializeAllTargets()':
/mnt/ccnas2/bdp/rz3515/projects/Polygeist/build/include/llvm/Config/Targets.def:26: undefined reference to `LLVMInitializeX86Target'
tools/polygeist/mlir-clang/CMakeFiles/mlir-clang.dir/__/llvm-project/clang/tools/driver/cc1_main.cpp.o: In function `llvm::InitializeAllTargetMCs()':
/mnt/ccnas2/bdp/rz3515/projects/Polygeist/build/include/llvm/Config/Targets.def:26: undefined reference to `LLVMInitializeX86TargetMC'
tools/polygeist/mlir-clang/CMakeFiles/mlir-clang.dir/__/llvm-project/clang/tools/driver/cc1_main.cpp.o: In function `llvm::InitializeAllAsmPrinters()':
/mnt/ccnas2/bdp/rz3515/projects/Polygeist/build/include/llvm/Config/AsmPrinters.def:27: undefined reference to `LLVMInitializeX86AsmPrinter'
tools/polygeist/mlir-clang/CMakeFiles/mlir-clang.dir/__/llvm-project/clang/tools/driver/cc1_main.cpp.o: In function `llvm::InitializeAllAsmParsers()':
/mnt/ccnas2/bdp/rz3515/projects/Polygeist/build/include/llvm/Config/AsmParsers.def:27: undefined reference to `LLVMInitializeX86AsmParser'
collect2: error: ld returned 1 exit status
ninja: build stopped: subcommand failed.
```
[CMakeCache.txt](https://github.com/wsmoses/Polygeist/files/7242962/CMakeCache.txt)

--- a/mlir-clang/CMakeLists.txt
+++ b/mlir-clang/CMakeLists.txt
@@ -1,17 +1,3 @@
-add_clang_tool(mlir-clang
-  "${LLVM_SOURCE_DIR}/../clang/tools/driver/cc1_main.cpp"
-  "${LLVM_SOURCE_DIR}/../clang/tools/driver/cc1as_main.cpp"
-  "${LLVM_SOURCE_DIR}/../clang/tools/driver/cc1gen_reproducer_main.cpp"
-  mlir-clang.cc
-  Lib/utils.cc
-  Lib/pragmaHandler.cc
-)
-
-target_include_directories(mlir-clang PRIVATE
-  "${LLVM_SOURCE_DIR}/../clang/include"
-  "${CMAKE_BINARY_DIR}/tools/clang/include"
-)
-
 set( LLVM_LINK_COMPONENTS
   ${LLVM_TARGETS_TO_BUILD}
   Analysis
@@ -28,16 +14,24 @@ set( LLVM_LINK_COMPONENTS
   Support
   TransformUtils
   Vectorize
-  )
+)
 
-list(TRANSFORM LLVM_LINK_COMPONENTS PREPEND LLVM)
+add_clang_tool(mlir-clang
+  "${LLVM_SOURCE_DIR}/../clang/tools/driver/cc1_main.cpp"
+  "${LLVM_SOURCE_DIR}/../clang/tools/driver/cc1as_main.cpp"
+  "${LLVM_SOURCE_DIR}/../clang/tools/driver/cc1gen_reproducer_main.cpp"
+  mlir-clang.cc
+  Lib/utils.cc
+  Lib/pragmaHandler.cc
+)
 
+target_include_directories(mlir-clang PRIVATE
+  "${LLVM_SOURCE_DIR}/../clang/include"
+  "${CMAKE_BINARY_DIR}/tools/clang/include"
+)
 
 target_compile_definitions(mlir-clang PUBLIC -DLLVM_OBJ_ROOT="${LLVM_BINARY_DIR}")
 target_link_libraries(mlir-clang PRIVATE
-    #    ${LLVM_LINK_COMPONENTS}
-    ${LLVM_AVAILABLE_LIBS}
-
   MLIRSCFTransforms
   MLIRPolygeist
 

